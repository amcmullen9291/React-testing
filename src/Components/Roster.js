import React, { useState } from 'react'

export default function Roster() {

    const students = [
        { id:1, firstName: "Alison Murphy", grade: 9, vehicle: "No Response"},
        { id:2, firstName: "Carl Jefferies", grade: 5, vehicle: "No Response" },
        { id:3, firstName: "Adam Marcus", grade: 3, vehicle: "No Response" },
        { id:4, firstName: "Lenny Tallerman", grade: 9, vehicle: "No Response" },
        { id:5, firstName: "Beth Rochester", grade: 12, vehicle: "No Response" },
    ];

    function switchAccess(e, id){
        e.preventDefault();
        let carInfo = document.getElementById(id);
       if((e.target.innerHTML == "Yes") || (e.target.innerHTML == "No Response") ){
        e.target.innerHTML = "No";
        carInfo.style.color = "red";
        carInfo.style.textDecoration = "line-through";
        let student = students.filter(x => x.id == id);
        student[0].vehicle = "No";
        return student;
    }else{
        e.target.innerHTML = "Yes"
        carInfo.style.color = "blue";
        carInfo.style.textDecoration = "none";
        let student = students.filter(x => x.id == id);
        student[0].vehicle = "Yes";
        return student;
    }
}

  return (
      <>
    <div data-testid="rosterList-1">
        <div>Roster</div>
        {students && (
              <div><table>
                <thead>
                    <tr>
                        <th>name</th>
                        <th>grade</th>
                        <th>vehicle?</th>
                    </tr>
                </thead>
                <tbody>
                {students.map((student, id ) => (
                    <tr key={student.id}>
                    <td>{student.firstName}</td>
                    <td>{student.grade}</td>
                    <td><a onClick={(e) => {switchAccess(e, student.id)}} id={student.id}>{student.vehicle}</a></td>
                    </tr>
                ))}
                </tbody>
        </table>
              </div>
            )}
    </div>
    </> 
  )
}
