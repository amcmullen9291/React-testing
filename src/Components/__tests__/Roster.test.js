import { render, screen, cleanup } from '@testing-library/react';
import Roster from '../Roster';
import renderer from 'react-test-renderer';

afterEach (() => {
    cleanup();
})

test('test', () => {
    expect(true).toBe(true);
});

test('Should render Roster component', () => {
    render(<Roster/>);
    const rosterElement = screen.getByTestId('rosterList-1'); // syntax; finds data-testid="rosterList-1", NOT id=""
    expect(rosterElement).toBeInTheDocument();  //is the element on the DOM
    //other options...
    expect(rosterElement).toHaveTextContent('ster');
})

test('should render student name', () => {
    const student = { id:6, firstName: "Adam Marcus", grade: 3, vehicle: "No" };
    render(<Roster student={student}/>)
    const rosterElement = screen.getByTestId('rosterList-1');
    expect(rosterElement).toBeInTheDocument();
    expect(rosterElement).toHaveTextContent(student.firstName);
})

it('renders correctly', () => {

    let student = {id: 7, firstName: "Mark Paul", grade: 10, vehicle: "No"};
    const tree = renderer
      .create(<Roster student={student}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
    console.log(tree);
  });