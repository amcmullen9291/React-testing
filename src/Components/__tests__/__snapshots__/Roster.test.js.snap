// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  data-testid="rosterList-1"
>
  <div>
    Roster
  </div>
  <div>
    <table>
      <thead>
        <tr>
          <th>
            name
          </th>
          <th>
            grade
          </th>
          <th>
            vehicle?
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Alison Murphy
          </td>
          <td>
            9
          </td>
          <td>
            <a
              id={1}
              onClick={[Function]}
            >
              No Response
            </a>
          </td>
        </tr>
        <tr>
          <td>
            Carl Jefferies
          </td>
          <td>
            5
          </td>
          <td>
            <a
              id={2}
              onClick={[Function]}
            >
              No Response
            </a>
          </td>
        </tr>
        <tr>
          <td>
            Adam Marcus
          </td>
          <td>
            3
          </td>
          <td>
            <a
              id={3}
              onClick={[Function]}
            >
              No Response
            </a>
          </td>
        </tr>
        <tr>
          <td>
            Lenny Tallerman
          </td>
          <td>
            9
          </td>
          <td>
            <a
              id={4}
              onClick={[Function]}
            >
              No Response
            </a>
          </td>
        </tr>
        <tr>
          <td>
            Beth Rochester
          </td>
          <td>
            12
          </td>
          <td>
            <a
              id={5}
              onClick={[Function]}
            >
              No Response
            </a>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
